apscheduler/executors/asyncio.py

```diff
@@ -31,29 +43,54 @@ def callback(f):
             else:
                 self._run_job_success(job.id, events)
 
-        events = self._job_runtime(job, run_times)
-        future_events = []
-        for event in events:
-            if not (isinstance(event, asyncio.Future) or asyncio.iscoroutine(event)):
-                future = asyncio.Future()
-                future.set_result(event)
-                future_events.append(future)
-            else:
-                future_events.append(event)
-        future = asyncio.gather(*events)
+        if not asyncio.iscoroutinefunction(job.func):
+            future = self._eventloop.run_in_executor(None, job_runtime, job, run_times, self._logger.name)
+        else:
+            events = job_runtime(job, run_times, self._logger.name, self._run_job)
+            future_events = []
+            for event in events:
+                if not (isinstance(event, asyncio.Future) or asyncio.iscoroutine(event)):
+                    future = asyncio.Future()
+                    future.set_result(event)
+                    future_events.append(future)
+                else:
+                    future_events.append(event)
+            future = asyncio.gather(*events)
         future.add_done_callback(callback)
 
-    @asyncio.coroutine
-    def _run_job(self, job, run_time):
-        """Actual implementation of calling the job function"""
-        try:
-            retval = yield from job.func(*job.args, **job.kwargs)
-        except:
-            exc, tb = sys.exc_info()[1:]
-            formatted_tb = ''.join(format_tb(tb))
-            self._logger.exception('Job "%s" raised an exception', job)
-            return JobExecutionEvent(EVENT_JOB_ERROR, job.id, job._jobstore_alias, run_time,
-                                     exception=exc, traceback=formatted_tb)
-        else:
-            self._logger.info('Job "%s" executed successfully', job)
-            return JobExecutionEvent(EVENT_JOB_EXECUTED, job.id, job._jobstore_alias, run_time, retval=retval)
+    if real_asyncio:
```

Comment:
> Native coroutines have the __await__ method which returns an iterable. On Python < 3.4, you can just iterate over the generator directly:

The problem is that they don't accept return inside generators, I already tried and the function doesn't get to execute, that's why trollius has the raise Return()

> There is also an option of using the async() or ensure_future() function which should work nicely together with run_on_executor which also returns a future. You'll need to experiment a bit though!

The problem with these is that you need someone to read the future result, or it won't execute. ATM, with the future + callback it works smoothly